File: app/src/main/java/bonoj/me/athenaeum/root/AthenaeumApplication.java
package bonoj.me.athenaeum.root;

import android.app.Application;

public class AthenaeumApplication extends Application {
    private ApplicationComponent component;

    @Override
    public void onCreate() {
        super.onCreate();

        // needs to run once to generate it
        component = DaggerApplicationComponent.builder()
                .applicationModule(new ApplicationModule(this))
                .build();
    }

    public ApplicationComponent getApplicationComponent() {
        return component;
    }
}

File: app/src/main/java/bonoj/me/athenaeum/root/AthenaeumApplication.kt
package bonoj.me.athenaeum.root

import android.app.Application

class AthenaeumApplication : Application() {

    companion object {
        @JvmStatic
        lateinit var graph: ApplicationComponent
    }

    override fun onCreate() {
        super.onCreate()
        graph = DaggerApplicationComponent.builder().applicationModule(ApplicationModule(this)).build()
        graph.inject(this)
    }
}
